airflow/providers/databricks/sensors/databricks_partition.py

```diff
@@ -0,0 +1,225 @@
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+"""This module contains Databricks sensors."""
+
+from __future__ import annotations
+
+from datetime import datetime
+from typing import TYPE_CHECKING, Any, Callable, Sequence
+
+from databricks.sql.utils import ParamEscaper
+
+from airflow.compat.functools import cached_property
+from airflow.exceptions import AirflowException
+from airflow.providers.common.sql.hooks.sql import fetch_all_handler
+from airflow.providers.databricks.hooks.databricks_sql import DatabricksSqlHook
+from airflow.sensors.base import BaseSensorOperator
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+
+class DatabricksPartitionSensor(BaseSensorOperator):
+    """
+    Sensor to detect the presence of table partitions in Databricks.
+
+    :param databricks_conn_id: Reference to :ref:`Databricks
+        connection id<howto/connection:databricks>` (templated), defaults to
+        DatabricksSqlHook.default_conn_name.
+    :param sql_warehouse_name: Optional name of Databricks SQL warehouse. If not specified, ``http_path``
+        must be provided as described below, defaults to None
+    :param http_path: Optional string specifying HTTP path of Databricks SQL warehouse or All Purpose cluster.
+        If not specified, it should be either specified in the Databricks connection's
+        extra parameters, or ``sql_warehouse_name`` must be specified.
+    :param session_configuration: An optional dictionary of Spark session parameters. If not specified,
+        it could be specified in the Databricks connection's extra parameters, defaults to None
+    :param http_headers: An optional list of (k, v) pairs
+        that will be set as HTTP headers on every request. (templated).
+    :param catalog: An optional initial catalog to use.
+        Requires Databricks Runtime version 9.0+ (templated), defaults to ""
+    :param schema: An optional initial schema to use.
+        Requires Databricks Runtime version 9.0+ (templated), defaults to "default"
+    :param table_name: Name of the table to check partitions.
+    :param partitions: Name of the partitions to check.
+        Example: {"date": "2023-01-03", "name": ["abc", "def"]}
+    :param partition_operator: Optional comparison operator for partitions, such as >=.
+    :param handler: Handler for DbApiHook.run() to return results, defaults to fetch_all_handler
+    :param client_parameters: Additional parameters internal to Databricks SQL connector parameters.
+    """
+
+    template_fields: Sequence[str] = (
+        "databricks_conn_id",
+        "schema",
+        "http_headers",
```

Comment:
I would bundle the table related identifiers together, in "catalog", "schema" and "table_name" order, followed by the partitions argument. Probably above http_headers.

Reply:
Changed.